rust/processor/src/db/postgres/migrations/2024-08-13-014340_add_latest_market_state/up.sql

```diff
@@ -0,0 +1,80 @@
+-- Your SQL goes here
+
+CREATE TABLE market_latest_state_event (
+  -- Transaction metadata.
+  transaction_version BIGINT NOT NULL,
+  sender VARCHAR(66) NOT NULL,
+  entry_function VARCHAR(200),
+  transaction_timestamp TIMESTAMP NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+
+  -- Market and state metadata.
+  market_id BIGINT NOT NULL,
+  symbol_bytes BYTEA NOT NULL,
+  bump_time TIMESTAMP NOT NULL, -- Note that bump and emit time are interchangeable.
+  market_nonce BIGINT NOT NULL,
+  trigger triggers NOT NULL,
+
+  -- State event data.
+  clamm_virtual_reserves_base BIGINT NOT NULL,
+  clamm_virtual_reserves_quote BIGINT NOT NULL,
+  cpamm_real_reserves_base BIGINT NOT NULL,
+  cpamm_real_reserves_quote BIGINT NOT NULL,
+  lp_coin_supply NUMERIC NOT NULL,
+  cumulative_stats_base_volume NUMERIC NOT NULL,
+  cumulative_stats_quote_volume NUMERIC NOT NULL,
+  cumulative_stats_integrator_fees NUMERIC NOT NULL,
+  cumulative_stats_pool_fees_base NUMERIC NOT NULL,
+  cumulative_stats_pool_fees_quote NUMERIC NOT NULL,
+  cumulative_stats_n_swaps BIGINT NOT NULL,
+  cumulative_stats_n_chat_messages BIGINT NOT NULL,
+  instantaneous_stats_total_quote_locked BIGINT NOT NULL,
+  instantaneous_stats_total_value_locked NUMERIC NOT NULL,
+  instantaneous_stats_market_cap NUMERIC NOT NULL,
+  instantaneous_stats_fully_diluted_value NUMERIC NOT NULL,
+  last_swap_is_sell BOOLEAN NOT NULL,
+  last_swap_avg_execution_price_q64 NUMERIC NOT NULL,
+  last_swap_base_volume BIGINT NOT NULL,
+  last_swap_quote_volume BIGINT NOT NULL,
+  last_swap_nonce BIGINT NOT NULL,
+  last_swap_time TIMESTAMP NOT NULL,
+  
+  -- Querying all post-bonding curve markets. i.e., markets with liquidity pools.
+  daily_tvl_per_lp_coin_growth_q64 NUMERIC NOT NULL,
+  in_bonding_curve BOOLEAN NOT NULL,
+  volume_in_1m_state_tracker NUMERIC NOT NULL,
+
+  PRIMARY KEY (market_id)
+);
+
+CREATE INDEX mkts_in_bonding_curve_idx
+ON market_latest_state_event (in_bonding_curve, market_id, market_nonce DESC)
+WHERE in_bonding_curve = TRUE;
```

Comment:
You do not need to include `in_bonding_curve` in the index since you only have `in_bonding_cruve = TRUE` anyway.

Reply:
The order does matter, but here, since you put `WHERE in_bonding_curve = TRUE` at the end, there is no point in indexing by `in_bonding_curve` a set of elements where `in_bonding_curve` is always true.

Reply:
So when I checked, I found that in the case where we aren't querying the entire row it doesn't make a difference, but if we're only getting `in_bonding_curve, market_id` it can be an `index-only` scan and is more efficient

Also just to note, I removed the `market_nonce DESC` since it's pointless to have when there's only one state event per market in this table:

```sql
CREATE INDEX mkts_in_bonding_curve_idx
ON market_latest_state_event (in_bonding_curve, market_id)
WHERE in_bonding_curve = TRUE;
```

Reply:
Changes here:

https://github.com/econia-labs/aptos-indexer-processors/pull/34/commits/d65d01b0a7b70592e5b897ec39669b7af45bbe92

Reply:
I'm still not sure about this.

I don't see why you would query `in_bonding_curve`. I mean if you have:

```sql
SELECT market_id
FROM maket_latest_state_event
WHERE in_bonding_curve = TRUE;
```

you do not need to also query `in_bonding_curve`, you know it's true.

So having an index ONLY on `market_id` would still be an index-only scan, right ?